pinot-common/src/main/java/org/apache/pinot/common/Closeables.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Arrays;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A utility class to help release multiple {@link Closeable}
+ * resources in a safe manner without leaking.
+ * As an example if we have a list of Closeable resources,
+ * then the following code is prone to leaking 1 or more
+ * subsequent resources if an exception is thrown while
+ * closing one of them.
+ *
+ * for (closeable_resource : resources) {
+ *   closeable_resource.close()
+ * }
+ *
+ * The helper methods provided here do this safely
+ * while keeping track of exception(s) raised during
+ * close() of each resource and still continuing to close
+ * subsequent resources.
+ */
+public class Closeables {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(Closeables.class);
+
+  /**
+   * Close a collection of {@link Closeable} resources
+   * @param closeables collection of resources to close
+   * @throws IOException
+   */
+  public static void close(Iterable<? extends Closeable> closeables)
```

Comment:
We have one class with one static function. Can we move this static function to FileUtils? I understand not all closeables are files ...

Reply:
We can move this to common/utils in Fileutils and probably rename FileUtils to IOUtils. 

In future, I think we should consider making these classes implement AutoCloseable instead of Closeable as that gives us flexibility of handling exceptions other than IOException. For example a close() on PinotDataBuffer might throw unpredictable exceptions as we release native memory. Once we move to AutoCloseable, then I think we should have these utility methods in their own class.